compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6742,9 +6747,23 @@ private BType checkIndexAccessExpr(BLangIndexBasedAccess indexBasedAccessExpr) {
                 return symTable.semanticError;
             }
 
+            if (varRefType.tag == TypeTags.UNION) {
```

Comment:
Shall we add a comment explaning what's being done here.

Reply:
This section is taken out in #28688 so will it be fine if we leave it as it is

Reply:
+1